File: app/src/main/java/me/evrooij/groceries/data/Account.java
package me.evrooij.groceries.data;

import org.parceler.Parcel;

/**
 * Author: eddy
 * Date: 27-11-16.
 */

@Parcel
public class Account {
    int id;
    String username;
    String email;
    String password;

    public Account(int id, String username, String email) {
        this.id = id;
        this.username = username;
        this.email = email;
    }

    public Account() {
    }

    public int getId() {
        return id;
    }

    public String getUsername() {
        return username;
    }

    public String getEmail() {
        return email;
    }

    @Override
    public String toString() {
        return String.format("User %s (%s) - Mail %s - Pass %s", username, id, email, password);
    }
}

File: app/src/main/java/me/evrooij/groceries/data/Account.kt
package me.evrooij.groceries.data

import android.os.Parcel
import android.os.Parcelable

/**
 * Author: eddy
 * Date: 27-11-16.
 */

data class Account(var id: Int, var username: String, var email: String) : Parcelable {
    companion object {
        @JvmField val CREATOR: Parcelable.Creator<Account> = object : Parcelable.Creator<Account> {
            override fun createFromParcel(source: Parcel): Account = Account(source)
            override fun newArray(size: Int): Array<Account?> = arrayOfNulls(size)
        }
    }

    constructor(source: Parcel) : this(source.readInt(), source.readString(), source.readString())

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel?, flags: Int) {
        dest?.writeInt(id)
        dest?.writeString(username)
        dest?.writeString(email)
    }
}